src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -82,17 +81,19 @@ public abstract class FastTreeTrainerBase<TArgs, TPredictor> :
 
         protected string InnerArgs => CmdParser.GetSettings(Host, Args, new TArgs());
 
-        public override bool NeedNormalization => false;
-
-        public override bool WantCaching => false;
+        public override TrainerInfo Info { get; }
 
         public bool HasCategoricalFeatures => Utils.Size(CategoricalFeatures) > 0;
 
-        protected internal FastTreeTrainerBase(IHostEnvironment env, TArgs args)
+        private protected FastTreeTrainerBase(IHostEnvironment env, TArgs args)
             : base(env, RegisterName)
         {
             Host.CheckValue(args, nameof(args));
             Args = args;
+            // The discretization step renders this trainer non-parametric, and therefore it does not need normalization.
+            // Also since it builds its own internal discretized columnar structures, it cannot benefit from caching.
+            // Finally, even the binary classifiers, being logitboost, tend to not benefit from external calibration.
+            Info = new TrainerInfo(normalization: false, caching: false, calibration: this is FastForestClassification);
```

Comment:
> `calibration: this is FastForestClassification`

It feels like that concrete classes should be able to decide this.  The base class shouldn't decide which of its concrete classes should have this true/false.

Reply:
Same comment for the GamTrainer.

---
In reply to: [202365055](https://github.com/dotnet/machinelearning/pull/522#discussion_r202365055) [](ancestors = 202365055)

Reply:
Fair enough